Explorer/Assets/DCL/Backpack/BackpackGridController.cs

```diff
@@ -0,0 +1,157 @@
+        {
+            world = builder.World;
+        }
+
+        public async UniTask InitialiseAssetsAsync(IAssetsProvisioner assetsProvisioner, CancellationToken ct)
+        {
+            BackpackItemView backpackItem = (await assetsProvisioner.ProvideInstanceAsync(view.BackpackItem, ct: ct)).Value;
+
+            gridItemsPool = new ObjectPool<BackpackItemView>(
+                () => CreateBackpackItem(backpackItem),
+                _ => { },
+                defaultCapacity: 16
+            );
+        }
+
+        private void SetGridElements(IWearable[] gridWearables)
+        {
+            ClearPoolElements();
+            for (var i = 0; i < gridWearables.Length; i++)
+            {
+                BackpackItemView backpackItemView = gridItemsPool.Get();
+                usedPoolItems.Add(gridWearables[i].GetUrn(), backpackItemView);
+
+                backpackItemView.OnSelectItem += SelectItem;
+                backpackItemView.ItemId = gridWearables[i].GetUrn();
+                backpackItemView.RarityBackground.sprite = rarityBackgrounds.GetTypeImage(gridWearables[i].GetRarity());
+                backpackItemView.FlapBackground.color = rarityColors.GetColor(gridWearables[i].GetRarity());
+                backpackItemView.CategoryImage.sprite = categoryIcons.GetTypeImage(gridWearables[i].GetCategory());
+
+                WaitForThumbnailAsync(gridWearables[i], backpackItemView).Forget();
+            }
+        }
+
+        private BackpackItemView CreateBackpackItem(BackpackItemView backpackItem)
+        {
+            BackpackItemView backpackItemView = Object.Instantiate(backpackItem, view.gameObject.transform);
+            backpackItem.EquipButton.onClick.AddListener(() => commandBus.SendCommand(new BackpackEquipCommand(backpackItemView.ItemId)));
+            return backpackItemView;
+        }
+
+        public void RequestPage(int pageNumber)
+        {
+            //Reuse params array and review types URLParameter once auth pr is merged
+            ParamPromise wearablesPromise = ParamPromise.Create(world, new GetWearableByParamIntention(new[] { ("pageNumber", string.Format("{0}", pageNumber)), ("pageSize", "16") },  web3IdentityCache.Identity!.EphemeralAccount.Address, new List<IWearable>()), PartitionComponent.TOP_PRIORITY);
+            AwaitWearablesPromiseAsync(wearablesPromise).Forget();
+        }
+
+        private async UniTaskVoid AwaitWearablesPromiseAsync(ParamPromise wearablesPromise)
+        {
+            AssetPromise<IWearable[],GetWearableByParamIntention> uniTaskAsync = await wearablesPromise.ToUniTaskAsync(world);
+
+            if (!uniTaskAsync.Result.Value.Succeeded)
+                return;
+
+            SetGridElements(uniTaskAsync.Result.Value.Asset);
+        }
+
+        private async UniTaskVoid WaitForThumbnailAsync(IWearable itemWearable, BackpackItemView itemView)
+        {
+            do
```

Comment:
This pattern is being repeated several times: I propose to introduce an extension to wait for `StreamableLoadingResult<T>?`